common/djangoapps/track/views/segmentio.py

```diff
@@ -163,7 +163,7 @@ def track_segmentio_event(request):  # pylint: disable=too-many-statements
     if 'application' not in app_context:
         context['application'] = {
             'name': app_context.get('app_name', ''),
-            'version': '' if not segment_context else segment_context.get('app', {}).get('version', '')
+            'version': segment_context.get('app', {}).get('version', ''),
```

Comment:
Instead of adding this special casing everywhere `segment_context` is indexed, let's just ensure it's always a dictionary, even if empty.